### PR TITLE
fix(FEC-7935): fix text selection logic with hlsjs

### PR DESCRIPTION
**Pull request by OrenMe**
### Description of the Changes

this is basically the same as before, but hls.js run different code paths when we use it's API for selecting text track.
Before this change we used to set the `showing`/`hidden` mode attribute of the text track in playkit-js html5 engine. we now set this in HLS.JS which internally does the same but also it manages internal state for this and so it's better to use its API and not access the textTrack.mode directly.
This fix depends on a fix in https://github.com/kaltura/playkit-js/pull/263/

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: test/src/hls-adapter.spec.js [hunk 1]
```diff
@@ -210,29 +210,6 @@ describe('HlsAdapter Instance - Unit', function () {
...
-      textTracks: hls_tracks.subtitles,
-      removeEventListener: () => {}
-    };
-    hlsAdapterInstance._disableAllTextTracks();
-    for (let i = 0; i < hlsAdapterInstance._videoElement.textTracks.length; i++) {
-      hlsAdapterInstance._videoElement.textTracks[i].mode.should.equal('disabled');
-    }
-  });
-
-  it('should hide the active text track', function () {
```
Comment (yairans) [inline test/src/hls-adapter.spec.js]: why to remove this test?

File: src/hls-adapter.js [hunk 2]
```diff
@@ -149,6 +149,9 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
         adapterConfig.hlsConfig.startPosition = config.playback.startTime;
       }
     }
+    if (Utils.Object.hasPropertyPath(config, 'playback.useNativeTextTrack')) {
+      adapterConfig.subtitleDisplay = Utils.Object.getPropertyPath(config, 'playback.useNativeTextTrack');
```
Comment (yairans) [inline src/hls-adapter.js]: just add it on `adapterConfig.hlsConfig`.

File: src/hls-adapter.js [hunk 3]
```diff
@@ -202,6 +205,7 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
       this._config.hlsConfig['pLoader'] = pLoader;
     }
     this._hls = new Hlsjs(this._config.hlsConfig);
+    this._hls.subtitleDisplay = this._config.subtitleDisplay;
```
Comment (yairans) [inline src/hls-adapter.js]: unnecessary. see the comment above.

File: src/hls-adapter.js (hunk 2 above)
Comment (OrenMe) [reply src/hls-adapter.js]: It's not an hls.js config option - it's an API. I wanted to add it as config first but it doesn't get set on hls.js config...

File: test/src/hls-adapter.spec.js (hunk 1 above)
Comment (OrenMe) [reply test/src/hls-adapter.spec.js]: Previously we managed the tracks directly ourselves - so it made sense.
Now we use hlsjs API and it requires loading an actual manifest with text tracks - and we will basically be testing internals of hlsjs - which is out of scope for unit testing

File: src/hls-adapter.js (hunk 3 above)
Comment (OrenMe) [reply src/hls-adapter.js]: see my comment above

File: test/src/hls-adapter.spec.js (hunk 1 above)
Comment (yairans) [reply test/src/hls-adapter.spec.js]: we just checkinh that `hideTextTrack` does its work

Comment (OrenMe) [reply test/src/hls-adapter.spec.js]: I understand but the test doesn't reflect it - in our unit testing we inject the text tracks ourselves and not via hlsjs so we actually don't test correctly.
And now that we moved to using hlsjs API itself then the test is both irrelevant(as the testing is black boxed behind hls.js API) and not working as we are not using hlsjs to add and manage the text tracks in the test itself.

**Pull request merged**